service/controller/azurecluster/handler/azureconfig/create.go

```diff
@@ -151,6 +152,9 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 			}
 		}
 
+		// Ensure External IP address annotation is up to date.
+		presentAzureConfig.Annotations[localannotation.WorkersEgressExternalPublicIP] = mappedAzureConfig.Annotations[localannotation.WorkersEgressExternalPublicIP]
```

Comment:
I prefer to have the annotation explicit. I don't want to copy over annotations I don't know about because, unlike labels, annotations most often have a direct effect on the cluster